### PR TITLE
Add excluded_field_kwargs to support custom OneToOneFields.

### LINKED ISSUE #870: Support custom OneToOneFields with additional parameters.
**Problem Statement**
Coercing `OneToOneField`s to be `ForeignKey`s on historical models causes issues when the `OneToOneField` is a custom subclass that includes additional arguments that do not exist on `ForeignKey.__init__`.

**Describe the solution you'd like**
In my particular case, ignoring the value would work well. My proposed solution is to supply a dictionary to `HistoricalRecords` that maps fields to the list of kwargs that should be excluded.

**Describe alternatives you've considered**
I did not consider alternatives.

**Additional context**
Currently when trying to use a custom `OneToOneField`, a `TypeError` is raised:

```
  File "python3.9/site-packages/simple_history/models.py", line 159, in finalize
    history_model = self.create_history_model(sender, inherited)
  File "python3.9/site-packages/simple_history/models.py", line 221, in create_history_model
    fields = self.copy_fields(model)
  File "python3.9/site-packages/simple_history/models.py", line 302, in copy_fields
    field = FieldType(*args, **field_args)
  File "python3.9/site-packages/django/db/models/fields/related.py", line 825, in __init__
    super().__init__(
  File "python3.9/site-packages/django/db/models/fields/related.py", line 474, in __init__
    super().__init__(rel=rel, **kwargs)
TypeError: __init__() got an unexpected keyword argument 'custom_field'
```

This is caused by #167 and somewhat related to #462.

### PR DESCRIPTION
**Pull request by tim-schilling**
Support custom OneToOneFields that specify additional arguments that don't exist in `ForeignKey.__init__`.

## Description

These additional arguments can be specified via excluded_field_kwargs to allow these custom OneToOneFields be coerced into ForeignKeys on the historical model.

This adds the `excluded_field_kwargs` parameter to `HistoricalRecords`. It is a dictionary mapping fields to list of parameters that should be excluded when crafting the ForeignKey instance for the historical model.

## Related Issue
Fixes #870

To reproduce, the following model can be added and then attempt to run `makemigrations` and `migrate`.

```
class CustomAttrNameOneToOneField(models.OneToOneField):
    def __init__(self, *args, **kwargs):
        self.attr_name = kwargs.pop("attr_name", None)
        super(CustomAttrNameOneToOneField, self).__init__(*args, **kwargs)

    def get_attname(self):
        return self.attr_name or super(CustomAttrNameOneToOneField, self).get_attname()

    def deconstruct(self):
        name, path, args, kwargs = super(
            CustomAttrNameOneToOneField, self
        ).deconstruct()
        if self.attr_name:
            kwargs["attr_name"] = self.attr_name
        return name, path, args, kwargs


class ModelWithCustomAttrOneToOneField(models.Model):
    poll = CustomAttrNameOneToOneField(Poll, models.CASCADE, attr_name="custom_poll")
    history = HistoricalRecords(excluded_field_kwargs={"poll": set(["attr_name"])})
```

## Motivation and Context
Coercing OneToOneFields to be ForeignKeys on historical models causes issues when the OneToOneField is a custom subclass that includes additional arguments that do not exist on the `ForeignKey.__init__`  method.

## How Has This Been Tested?

The problem was able to be reproduced by adding `CustomAttrNameOneToOneField` and `ModelWithCustomAttrOneToOneField` to the test app. *There's an additional test that verifies the constraint of this change in that the additional arguments are not available on the ForeignKey field.* This is done via `TestCustomAttrOneToOneField`.

## Screenshots (if appropriate):
N/A

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have run the `make format` command to format my code
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] I have added my name and/or github handle to `AUTHORS.rst`
- [x] I have added my change to `CHANGES.rst`
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by tim-schilling** [comment]
I didn't check the tests case because I did not run the entire testing matrix locally.

File: simple_history/registry_tests/tests.py [hunk 1]
```diff
@@ -198,13 +199,21 @@ def test_registering_with_tracked_abstract_base(self):
 
 
 class TestCustomAttrForeignKey(TestCase):
-    """ https://github.com/jazzband/django-simple-history/issues/431 """
+    """https://github.com/jazzband/django-simple-history/issues/431"""
```
Comment (tim-schilling) [inline simple_history/registry_tests/tests.py:202]: This was a result of `make format`.

File: simple_history/registry_tests/tests.py [hunk 2]
```diff
@@ -219,7 +228,7 @@ def test_migrate_command(self):
 
 
 class TestModelWithHistoryInDifferentApp(TestCase):
-    """ https://github.com/jazzband/django-simple-history/issues/485 """
+    """https://github.com/jazzband/django-simple-history/issues/485"""
```
Comment (tim-schilling) [inline simple_history/registry_tests/tests.py:231]: This was a result of `make format`.

**Comment by tim-schilling** [comment]
And I named the branch incorrectly... It should have been 870, not 807.

**Review by jeking3 — APPROVED** [review]
Documentation and unit tests - looks good.  Need to rebase and fix up the collision in CHANGES.  If this gets in before my other two I will rebase them to fix the resulting migration collisions.

**Comment by jeking3** [comment]
@tim-schilling there's probably just one more branch that needs coverage here, and a rebase to fix up the changes file.

**Review by jeking3 — APPROVED** [review]

**Pull request merged**